src/screens/NewTerm.js

```diff
@@ -0,0 +1,201 @@
+import React, {useEffect, useState} from 'react';
+import {Picker} from '@react-native-picker/picker';
+import {action} from '@storybook/addon-actions';
+import 'react-native-get-random-values';
+import {v4 as uuid} from 'uuid';
+import {
+  Text,
+  View,
+  StyleSheet,
+  SafeAreaView,
+  KeyboardAvoidingView,
+} from 'react-native';
+
+import ToolBar from '../components/ToolBar/ToolBar';
+import Textarea from '../components/Textarea/Textarea';
+import NextButton from '../components/NextButton/NextButton';
+import ToolButton from '../components/ToolButton/ToolButton';
+import LanguageSwitcher from '../components/LanguageSwitcher/LanguageSwitcher';
+import BackIcon from '../components/ToolButton/assets/back.svg';
+import ModeIcon from '../components/ToolButton/assets/mode.svg';
+
+import {LANGUAGE_NAMES, getAllCategories} from '../data/dataUtils';
+import SectionHeading from '../components/SectionHeading/SectionHeading';
+
+export default ({
+  //nav provider
+  navigation,
+  categories,
+  setCategories,
+  addNewPhrases,
+}) => {
+  const [selectedCategory, setSelectedCategory] = useState('');
+  const [malagasyText, setMalagasyText] = useState('');
+  const [englishText, setEnglishText] = useState('');
+  const [categoryId, setCategoryId] = useState('');
+
+  // find the selected category
+  const findSelectedId = Text => {
+    setSelectedCategory(Text);
+    const selectedCat = categories.find(cat => cat.name.en === Text);
```

Comment:
This is Ok but the naming variable should be more descriptive in line 40 in scr/creen/NewTerm.js, this will hard for your team to understand what it really does. This variable is actually for the selected category.